03_url_shortener/handler/handlers.go

```diff
@@ -13,30 +14,57 @@ type UrlCreationRequest struct {
 	LongUrl string `json:"long_url" binding:"required"` // binding is specific to gin
 }
 
-func CreateShortUrl(c *gin.Context) {
+type HttpHandler struct {
+	storageService *store.StorageService
+}
+
+func NewHttpHandler(storage *store.StorageService) *HttpHandler {
+	return &HttpHandler{
+		storageService: storage,
+	}
+}
+
+func (h *HttpHandler) CreateShortUrl(c *gin.Context) {
 	var creationRequest UrlCreationRequest
 	// handle error
 	if err := c.ShouldBindJSON(&creationRequest); err != nil {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		log.Infof("error binding JSON: %q", err.Error())
+		// adding a return will stop the function from continuing
 		return
 	}
 
-	shortUrl := shortener.GenerateShortLink(creationRequest.LongUrl)
-	store.SaveUrlMapping(shortUrl, creationRequest.LongUrl)
+	shortUrl, err := shortener.GenerateShortLink(creationRequest.LongUrl)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		log.Infof("error generating short URL: %q", err.Error())
+		return
+	}
+	err = h.storageService.SaveUrlMapping(shortUrl, creationRequest.LongUrl)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		log.Infof("error saving URL to database: %q", err.Error())
+		return
+	}
 
 	host := "http://localhost:9808/"
-	c.JSON(200, gin.H{
+	c.JSON(http.StatusOK, gin.H{
 		"message":   "short url created successfully",
 		"short_url": host + shortUrl,
 	})
```

Comment:
Please read the `host` from config, since you will use some domain name in production.

Reply:
The host from config in this case is the host for the redis DB. Would it be good practice to include an application host in the .env as well?